### PR TITLE
Add Google Cloud Storage S3 compatible table function

### PR DESCRIPTION
**Pull request by jkaflik**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add Google Cloud Storage S3 compatible table function `gcs`. Like the `oss` and `cosn` functions, it is just an alias over the `s3` table function, and it does not bring any new features. 

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/TableFunctions/TableFunctionS3.h [hunk 1]
```diff
@@ -93,4 +93,26 @@ class TableFunctionOSS : public TableFunctionS3
 
 }
 
+class TableFunctionGCS : public TableFunctionS3
```
[the remark this answers is not in the document]
Comment (jkaflik) [reply src/TableFunctions/TableFunctionS3.h:96]: Thanks @nikitamikhaylov .

Comment (alexey-milovidov) [inline src/TableFunctions/TableFunctionS3.h:96]: You've put it outside of the namespace.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 0672d75f72099f0043757bb45830cbc1e24f5ba8 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/47815/0672d75f72099f0043757bb45830cbc1e24f5ba8/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by jkaflik** [comment]
@alexey-milovidov I added inline code documentation. I did it for S3, also. Not familiar with COS and OSS. It can be added in the future.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**